shell/platform/darwin/macos/framework/Source/FlutterTextInputPlugin.mm

```diff
@@ -51,6 +51,143 @@
 static NSString* const kTextKey = @"text";
 static NSString* const kTransformKey = @"transform";
 
+static NSString* const kDeleteCharacterIntent = @"TextInputClient.DeleteCharacterIntent";
+static NSString* const kDeleteToNextWordBoundaryIntent =
+    @"TextInputClient.DeleteToNextWordBoundaryIntent";
+static NSString* const kDeleteToLineBreakIntent = @"TextInputClient.DeleteToLineBreakIntent";
+static NSString* const kExtendSelectionByCharacterIntent =
+    @"TextInputClient.ExtendSelectionByCharacterIntent";
+static NSString* const kExtendSelectionToNextWordBoundaryIntent =
+    @"TextInputClient.ExtendSelectionToNextWordBoundaryIntent";
+static NSString* const kExtendSelectionToNextWordBoundaryOrCaretLocationIntent =
+    @"TextInputClient.ExtendSelectionToNextWordBoundaryOrCaretLocationIntent";
+static NSString* const kExtendSelectionToLineBreakIntent =
+    @"TextInputClient.ExtendSelectionToLineBreakIntent";
+static NSString* const kExtendSelectionVerticallyToAdjacentLineIntent =
+    @"TextInputClient.ExtendSelectionVerticallyToAdjacentLineIntent";
+static NSString* const kExtendSelectionToDocumentBoundaryIntent =
+    @"TextInputClient.ExtendSelectionToDocumentBoundaryIntent";
+static NSString* const kPerformPrivateCommandIntent =
+    @"TextInputClient.PerformPrivateTextInputCommandIntent";
+
+@interface SelectorIntent : NSObject
+@property(nonatomic, readonly) SEL selector;
+@property(nonatomic, readonly) NSArray* methodCall;
+
+- (instancetype)initWithSel:(SEL)selector methodCall:(NSArray<NSObject*>*)methodCall;
+@end
+
+@implementation SelectorIntent
+
+- (instancetype)initWithSel:(SEL)selector methodCall:(NSArray<NSObject*>*)methodCall {
+  self = [super init];
+  if (self) {
+    _selector = selector;
+    _methodCall = methodCall;
+  }
+  return self;
+}
+
++ (instancetype)intentWith:(SEL)selector methodCall:(NSArray<NSObject*>*)methodCall {
+  return [[SelectorIntent alloc] initWithSel:selector methodCall:methodCall];
+}
+
+@end
+
+static NSArray<SelectorIntent*>* const kIntentLookUp = @[
+  // Delete
+  [SelectorIntent intentWith:@selector(deleteBackward:)
+                  methodCall:@[ kDeleteCharacterIntent, @(false) ]],
```

Comment:
At first blush it strikes me as awkward that we're sort of remotely calling these intents like this, passing parameters without compile time error checking like we would have in the framework.  I can't think of a better way though.  We shouldn't send deleteBackward etc. to the framework because then we're mixing MacOS concerns with the framework.

Reply:
Yeah I think I was originally straight up sending the selector names as strings to the framework then the framework would have to translate macOS selector names to intents. So following our convention I'm doing the translation in the text input plugin and the framework just have to "reify" the intent. The platform messages are sent in json anyway so I feel there isn't much we can get by leveraging dart's type system.